[Runtime Metrics] Time in gc metric (#643)
signalfx/signalfx-dotnet-tracing: docs/metrics.md

The names and the metric structure of the metrics exported are aligned with OpenTelemetry
[implementation](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/bc947a00c3f859cc436f050e81172fc1f8bc09d7/src/OpenTelemetry.Instrumentation.Runtime).

| Metric                                                 | Description                                                                                                                     | Type              |

> |:-------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------|:------------------|
> | `process.runtime.dotnet.exceptions.count`              | The count of exceptions that have been thrown in managed code since the previous observation.                                   | Counter           |
> | `process.runtime.dotnet.gc.collections.count`          | The number of garbage collections that have occurred since the process was started.                                             | CumulativeCounter |
> | `process.runtime.dotnet.gc.heap.size`                  | The heap size, as observed during the latest garbage collection.                                                                | Gauge             |
> | `process.runtime.dotnet.gc.allocations.size`           | The count of bytes allocated on the managed GC heap since the process was started. (.NET Core only)                             | CumulativeCounter |
> | `process.runtime.dotnet.gc.committed_memory.size`      | The amount of committed virtual memory for the managed GC heap, as observed during the latest garbage collection. (.NET 6 only) | Gauge             |
> | `process.runtime.dotnet.gc.pause.time`                 | The number of milliseconds spent in GC pause. (.NET Core only)                                                                  | Counter           |
> | `process.runtime.dotnet.monitor.lock_contention.count` | The number of times there was contention when trying to acquire a monitor lock since the process was started.                   | CumulativeCounter |
> | `process.runtime.dotnet.thread_pool.threads.count`     | The number of thread pool threads, as observed during the latest measurement. (.NET Core only)                                  | Gauge             |

Reviewer: ```suggestion
|:-------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------|:------------------|
| `process.runtime.dotnet.exceptions.count`              | Count of exceptions since the previous observation.                                   | Counter           |
| `process.runtime.dotnet.gc.collections.count`          | Number of garbage collections since the process started.                                             | CumulativeCounter |
| `process.runtime.dotnet.gc.heap.size`                  | Heap size, as observed during the last garbage collection.                                                                | Gauge             |
| `process.runtime.dotnet.gc.allocations.size`           | Count of bytes allocated on the managed GC heap since the process started. (.NET Core only)                             | CumulativeCounter |
| `process.runtime.dotnet.gc.committed_memory.size`      | Amount of committed virtual memory for the managed GC heap, as observed during the last garbage collection. (.NET 6 only) | Gauge             |
| `process.runtime.dotnet.gc.pause.time`                 | Number of milliseconds spent in GC pause. (.NET Core only)                                                                  | Counter           |
| `process.runtime.dotnet.monitor.lock_contention.count` | Contentions count when trying to acquire a monitor lock since the process started.                   | CumulativeCounter |
| `process.runtime.dotnet.thread_pool.threads.count`     | Number of thread pool threads, as observed during the last measurement. Only available for .NET Core.                                  | Gauge             |
```
